docs/ref-bundle-stages.md

```diff
@@ -17,3 +17,40 @@ To demonstrate the life cycle of a Fleet bundle, we will use [multi-cluster/helm
 This diagram shows the different rendering stages a bundle goes through until deployment.
 
 ![Bundle Stages](/img/FleetBundleStages.svg)
+
+## Examining the Bundle Lifecycle With the CLI
+
+Several fleet CLI commands help with debugging bundles.
+
+### fleet apply
+
+[Apply](./cli/fleet-cli/fleet_apply.md) renders a folder with Kubernetes resources, such as a Helm chart, manifests, or kustomize folders, into a Fleet bundle resource.
+
+```
+git clone https://github.com/rancher/fleet-test-data
+cd fleet-test-data
+fleet apply -n fleet-local -o bundle.yaml testbundle simple-chart/
+```
+
+More information on how to create bundles with `fleet apply` can be found in the [section on bundles](https://fleet.rancher.io/bundle-add).
+
+### fleet target
+
+[Target](./cli/fleet-cli/fleet_target.md) reads a bundle from a file and works with a live cluster to print out the `bundledeployment` & `content` resource, which fleetcontroller would create. It takes a namespace as an argument, so it can look in that namespace for e.g. cluster resources. It can also dump the data structure which is used during "targeting", so decisions taken regarding labels and cluster names can be checked.
+
+### fleet deploy
+
+[Deploy](./cli/fleet-cli/fleet_deploy.md) takes the output of `fleet target`, or a dumped bundledeployment/content resource and deploys it to a cluster, just like fleet-agent would. It supports a dry run mode, to print out the resources which would be created, instead of installing them with helm. Since the command doesn't create the input resources, a running fleet-agent would likely garbage collect the deployment.
```

Comment:
I think it could make sense to clarify what is meant by _input resources_, eg:
```suggestion
[Deploy](./cli/fleet-cli/fleet_deploy.md) takes the output of `fleet target`, or a dumped bundledeployment/content resource and deploys it to a cluster, just like fleet-agent would. It supports a dry run mode, to print out the resources which would be created, instead of installing them with helm. Since the command doesn't create the input resources (eg. bundles or `GitRepo`s), a running fleet-agent would likely garbage collect the deployment.
```